src/main/scala/com/databricks/spark/xml/parsers/StaxXmlParser.scala

```diff
@@ -214,7 +214,7 @@ private[xml] object StaxXmlParser {
     // The fields are sorted so `TreeMap` is used.
     val convertedValuesMap = convertValues(valuesMap, schema)
     val row = TreeMap((fields ++ convertedValuesMap).toSeq : _*).values.toSeq
-    Row.fromSeq(row)
+    if(row.length == 0) null else Row.fromSeq(row)
```

Comment:
I think `if (row.isEmpty) null else Row.fromSeq(row)` might be better.
Also, could you maybe add a comment why this checking is necessary?